devnet-sdk/devstack/sysext/helpers.go

```diff
@@ -23,31 +25,89 @@ const (
 	FeatureInterop = "interop"
 )
 
-func (orch *Orchestrator) rpcClient(t devtest.T, endpoint string) client.RPC {
+func (orch *Orchestrator) rpcClient(t devtest.T, service *descriptors.Service, protocol string) client.RPC {
+	t.Helper()
+
+	endpoint, header, err := orch.findProtocolService(service, protocol)
+	t.Require().NoError(err)
+
 	opts := []client.RPCOption{}
 	if !orch.useEagerRPCClients {
 		opts = append(opts, client.WithLazyDial())
 	}
+
+	if orch.env.ReverseProxyURL != "" && !orch.useDirectCnx {
+		opts = append(
+			opts,
+			client.WithGethRPCOptions(
+				rpc.WithHeaders(header),
+				// we need both Header["Host"] and req.Host to be set
+				rpc.WithHTTPClient(&http.Client{
+					Transport: &hostSettingRoundTripper{
+						host: header.Get("Host"),
+						rt:   http.DefaultTransport,
+					},
+				}),
+			),
+		)
```

Comment:
This block seems like a near duplicate of the one in `httpClient` - could we maybe encapsulate that logic in `httpClient` instead?

Reply:
done